test/gurka/test_trace_attributes.cc

```diff
@@ -150,4 +150,186 @@ TEST(Standalone, RetrieveTrafficSignal) {
+
+  api = gurka::do_action(valhalla::Options::trace_attributes, map, {"A", "B", "C"}, "auto",
+                         {{"/shape_match", "edge_walk"},
+                          {"/date_time/type", "1"},
+                          {"/date_time/value", "2025-06-27T08:00"},
+                          {"/trace_options/breakage_distance", "10000"}},
+                         {}, &trace_json, "via");
+  result.Parse(trace_json.c_str());
+  edges = result["edges"].GetArray();
+
+  for (int i = 0; i < edges.Size(); i++) {
+    EXPECT_FALSE(edges[i].HasMember("speeds_faded"));
+    EXPECT_TRUE(edges[i]["speeds_non_faded"].HasMember("current_flow"));
+    EXPECT_TRUE(edges[i]["speeds_non_faded"].HasMember("predicted_flow"));
+    EXPECT_TRUE(edges[i]["speeds_non_faded"].HasMember("constrained_flow"));
+    EXPECT_TRUE(edges[i]["speeds_non_faded"].HasMember("free_flow"));
+    EXPECT_TRUE(edges[i]["speeds_non_faded"].HasMember("no_flow"));
+  }
+
+  api =
+      gurka::do_action(valhalla::Options::trace_attributes, map, {"A", "B", "C"}, "auto",
+                       {{"/shape_match", "edge_walk"}, {"/trace_options/breakage_distance", "10000"}},
+                       {}, &trace_json, "via");
+  result.Parse(trace_json.c_str());
+  edges = result["edges"].GetArray();
+
+  for (int i = 0; i < edges.Size(); i++) {
+    EXPECT_FALSE(edges[i].HasMember("speeds_faded"));
+    EXPECT_FALSE(edges[i]["speeds_non_faded"].HasMember("predicted_flow"));
+    EXPECT_TRUE(edges[i]["speeds_non_faded"].HasMember("current_flow"));
+    EXPECT_TRUE(edges[i]["speeds_non_faded"].HasMember("constrained_flow"));
+    EXPECT_TRUE(edges[i]["speeds_non_faded"].HasMember("free_flow"));
+    EXPECT_TRUE(edges[i]["speeds_non_faded"].HasMember("no_flow"));
+  }
+
+  // reset historical traffic
+  test::customize_historical_traffic(map.config, [&](DirectedEdge& e) {
+    e.set_constrained_flow_speed(0);
+    e.set_free_flow_speed(0);
+
+    return std::nullopt;
+  });
+  // invalidate traffic speed
+  test::customize_live_traffic_data(map.config, [&](baldr::GraphReader&, baldr::TrafficTile&, int,
+                                                    valhalla::baldr::TrafficSpeed* traffic_speed) {
+    traffic_speed->overall_encoded_speed = UNKNOWN_TRAFFIC_SPEED_RAW;
+    traffic_speed->encoded_speed1 = UNKNOWN_TRAFFIC_SPEED_RAW;
+    traffic_speed->breakpoint1 = 0;
+  });
+
+  // this request should not have current, predicted, constrained nor free flows
+  api = gurka::do_action(valhalla::Options::trace_attributes, map, {"A", "B", "C"}, "auto",
+                         {{"/shape_match", "edge_walk"},
+                          {"/date_time/type", "0"},
+                          {"/date_time/value", "current"},
+                          {"/trace_options/breakage_distance", "10000"}},
+                         {}, &trace_json, "via");
+
+  result.Parse(trace_json.c_str());
+  edges = result["edges"].GetArray();
+
+  for (int i = 0; i < edges.Size(); i++) {
```

Comment:
```suggestion
  for (size_t i = 0; i < edges.Size(); i++) {
```

Reply:
IIRC even this is not correct typing with rapidjson and there'll be some warning when not doing
```suggestion
  for (rapidjson::SizeType i = 0; i < edges.Size(); i++) {
```

> RapidJSON uses 32-bit array/string indices even on 64-bit platforms,
    instead of using \c size_t. Users may override the SizeType by defining
    \ref RAPIDJSON_NO_SIZETYPEDEFINE.

:roll_eyes: